frontend/src/metabase-lib/lib/queries/utils/native-query.ts

```diff
@@ -0,0 +1,37 @@
+import slugg from "slugg";
+import _ from "underscore";
+
+import type NativeQuery from "../NativeQuery";
+
+export function updateReferencedQuestionTagName(
+  queryText: string,
+  question: any,
+  oldTagName: string,
+) {
+  const newTagName = `#${question.id}-${slugg(question.name)}`;
+  return queryText.replace(
+    new RegExp(`{{\\s*${oldTagName}\\s*}}`, "g"),
+    `{{${newTagName}}}`,
+  );
+}
+
+export function updateReferencedQuestionNames(
```

Comment:
Would be very meaningful to cover this function with unit tests.

Reply:
Yes, see https://github.com/metabase/metabase/commit/44d426e48c7620f12b1e93eca9a840c6a2e1767b
Note I renamed `updateReferencedQuestionNames` to `updateCardTagNames` for consistency